### PR TITLE
api: Fix search by name

### PR DESCRIPTION
**Pull request by Pearl1594**
### Description

This PR aims at restricting using name field in the list APIs to do exact matching as the APIs have a keyword field too. This is an extension of https://github.com/apache/cloudstack/pull/5782 - specifically addressing the following comment: https://github.com/apache/cloudstack/pull/5782#issuecomment-995786499

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [X] Minor

### CODE REVIEW

**Comment by Pearl1594** [comment]
@blueorangutan package

**Comment by Pearl1594** [comment]
@blueorangutan test

File: engine/schema/src/main/java/org/apache/cloudstack/acl/dao/ProjectRoleDaoImpl.java [hunk 1]
```diff
@@ -17,45 +17,56 @@
...
-        ProjectRoleSearch.and("name", ProjectRoleSearch.entity().getName(), SearchCriteria.Op.LIKE);
+        ProjectRoleSearch.and("name", ProjectRoleSearch.entity().getName(), SearchCriteria.Op.EQ);
         ProjectRoleSearch.and("project_id", ProjectRoleSearch.entity().getProjectId(), SearchCriteria.Op.EQ);
+        ProjectRoleSearch.and("keyword", ProjectRoleSearch.entity().getName(), SearchCriteria.Op.LIKE);
         ProjectRoleSearch.done();
 
     }
     @Override
-    public List<ProjectRoleVO> findByName(String name, Long projectId) {
+    public List<ProjectRoleVO> findByName(String name, Long projectId, String keyword) {
```
Comment (DaanHoogland) [inline engine/schema/src/main/java/org/apache/cloudstack/acl/dao/ProjectRoleDaoImpl.java:42]: this looks to accept both name and keyword and search for name adhering to both. Is that intended?

Comment (Pearl1594) [reply engine/schema/src/main/java/org/apache/cloudstack/acl/dao/ProjectRoleDaoImpl.java:42]: Name would take precedence @DaanHoogland , so it's effectively not going to use the keyword passes. I can actually remove it.

Comment (weizhouapache) [reply engine/schema/src/main/java/org/apache/cloudstack/acl/dao/ProjectRoleDaoImpl.java]: @DaanHoogland 
yes, in most list apis, keyword is acceptable with other parameters like name, type, etc.

for this particular api, I personally think it is not necessary to accept `keyword`
but, it is ok to me.

**Review by DaanHoogland — APPROVED** [review]
clgtm

File: engine/schema/src/main/java/org/apache/cloudstack/acl/dao/ProjectRoleDaoImpl.java [hunk 2]
```diff
@@ -17,45 +17,51 @@
...
 
     @Override
-    public List<ProjectRoleVO> findAllRoles(Long projectId) {
+    public List<ProjectRoleVO> findAllRoles(Long projectId, String keyword) {
         SearchCriteria<ProjectRoleVO> sc = ProjectRoleSearch.create();
         if (projectId != null) {
             sc.setParameters("project_id", projectId);
         }
+        if (StringUtils.isNotEmpty(keyword)) {
+            SearchCriteria<ProjectRoleVO> ssc = createSearchCriteria();
```
Comment (weizhouapache) [inline engine/schema/src/main/java/org/apache/cloudstack/acl/dao/ProjectRoleDaoImpl.java:61]: @Pearl1594 
why not use
```
sc.setParameters("keyword", keyword);
```

**Review by rohityadavcloud — APPROVED** [review]
Lgtm, we already have keyword param for fuzzy search

**Review by nvazquez — APPROVED** [review]

**Pull request merged**